### DISCUSSION TITLE
VulkanSceneGraph-1.0.9-rc1, vsgXchange-1.0.5-rc1 and vsgExamples-1.0.7-rc1 tagged, please test :-)

### DISCUSSION THREAD

**Discussion by robertosfield** [vsg-dev/VulkanSceneGraph#934] [General]
Hi All,

It been a month since the last stable release of the VSG and there is number useful refinements, new capabilities and thousands of typo fixes so I have tagged release candidates for the VSG, vsgXchange and vsgExamples in prep for new stable releases next week.  The release candidates are:

- https://github.com/vsg-dev/VulkanSceneGraph/releases/tag/v1.0.9-rc1
- https://github.com/vsg-dev/vsgXchange/releases/tag/v1.0.5-rc1
- https://github.com/vsg-dev/vsgExamples/releases/tag/v1.0.7-rc1

The typo fixes from @LeeRiva and @siystar are much appreciated and make up for having Dyslexic project lead :-) 

The major features additions since VulkanSceneGraph-1.0.8:

- Support for Masks in vsg::ShaderStage and vsg::GraphicsPipelinesState which enable easy pipeline specialization on different views.  I discuss this topic on forum thread: #916.
- Support for CommandGraph nesting and explict submission order control using the new CommandGraph::submitOrder. I discuss this topic on the forum thread: #929.

There are a couple of API changes that may break the build for you, if you have issues review topic #916 discussion and the links it provides to see how you might be able to address them.

These release candidates need testing so I would appreciate testing on all the platforms you have available, testing both vsgExamples and your own applications/libraries to make sure everything is building and running correctly.   Post reports of success and failure here on this thread so we can see how well the codebase is converging to something ready for release.  All going well I'll tag the official releases next week.

Thanks for you help,
Robert.

**Comment by timoore** [comment]
FYI, I have built and run vsgCs against master, so it's good to go from my point of view.

**Comment by robertosfield** [comment]
A regression in the vsg::Text billboard support crept into VSG master prior to merging the VulkanSceneGraph-1.0 branch and tagging 1.0.9-rc1.  I have fixed this in VSG master with the commit: 23c13fc98359a024bcfa3c87c4d44b8ca2dcd95f.

The vsgdynamicstate example is also not functioning correctly for me right now so I will investigate that next,

**Comment by robertosfield** [comment]
I have fixed the issue with the vsgdynamicstate example.  Seems drivers I used use let that example by even though not all the state was being enabled correctly: [Fix to vsgdynamicstate](https://github.com/vsg-dev/vsgExamples/commit/f6e1eefb0c91a05181a286bda267ed57fc457c1d).

**Comment by robertosfield** [comment]
To address the issues picked up and fixed in VulkanSceneGraph v1.0.9-rc1 and vsgExamples v1.0.7-rc1 I have tagged rc2's for each of these projects:

https://github.com/vsg-dev/VulkanSceneGraph/releases/tag/v1.0.9-rc2
https://github.com/vsg-dev/vsgExamples/releases/tag/v1.0.7-rc2

**Comment by robertosfield** [comment]
I have been away on a family holiday for a long weekend, I have just arrived back. Tomorrow I am back in my office and will review the PR's that have been made since Thursday when I tagged the rc's and merge where appropriate.  I'll likely then tag further rc's where appropriate.

**Comment by robertosfield** [comment]
I have now reviewed, merged and amended the PRs that were posted to VSG and vsgXchange since rc2 and have created the following release candidates:

https://github.com/vsg-dev/VulkanSceneGraph/releases/tag/v1.0.9-rc3
https://github.com/vsg-dev/vsgXchange/releases/tag/v1.0.5-rc2

One of the PRs (#939) I have reviewed and decided to close as while it fixes one issue it highlights another that it doesn't resolve, so i think the best thing to do is to conduct a wider review of how vsg::ImageInfo. vsg::Samplervsg:, vsg::ImageView and vsg::Image interact to resolve the problems in certain usages cases where different ImageInfo are used with different combinations of vsg::Sampler and vsg::Image.  I have created Issue #945 for this topic. 

Ideally I'd like to come up with a solution before 1.0.9 is released but I'll need to balance this work with other work that I need to get on with, essentially I can't let releases consume too much of my time as I'll have none left to do the client work that pays the bills.

**Comment by robertosfield** [comment]
I have spent the day look at how ImageInfo/imageView/Image/DescriptorImage handle the Image:mipLevels and Sample::maxLod settings and have checked in changes to make it work more user controllable and allow Sampler to be shared between ImageInfo without their maxLod value being modified.  The changes are now checked into with the PR: #950.

I have merged these changes into VSG master and VulkanSceneGraph-1.0 branch, and added a vsgvalidate example into vsgExamples so have updated it's master and vsgExamples-1.0 with this new example, and have created new release candidates:

https://github.com/vsg-dev/VulkanSceneGraph/releases/tag/v1.0.9-rc4
https://github.com/vsg-dev/vsgExamples/releases/tag/v1.0.7-rc3

Please test , and report success/failure on this thread. if all goes well I'll tag the final releases tomorrow or on Friday :-)

**Comment by robertosfield** [comment]
I am now ready to tag the releases.  I'll leave it another hour for folks to test and provide feedback, then if no issues have been raised I'll tag the releases.
